Tests fail in 1.18.0

Running `go test -v ./...` on version 1.18.0 gives failures on Arch Linux, although it was fine on 1.17.9:
```
=== RUN   TestGenerateSpec
=== RUN   TestGenerateSpec/default
    generate_test.go:460:
        	Error Trace:	/build/nvidia-container-toolkit/src/nvidia-container-toolkit-1.18.0/cmd/nvidia-ctk/cdi/generate/generate_test.go:460
        	Error:      	Not equal:
        	            	expected: generate.options{output:"", format:"yaml", deviceNameStrategies:[]string(nil), driverRoot:"/build/nvidia-container-toolkit/src/nvidia-container-toolkit-1.18.0/testdata/lookup/rootfs-1", devRoot:"", nvidiaCDIHookPath:"/usr/bin/nvidia-cdi-hook", ldconfigPath:"", mode:"nvml", vendor:"example.com", class:"device", configSearchPaths:[]string(nil), librarySearchPaths:[]string(nil), disabledHooks:[]string(nil), enabledHooks:[]string(nil), featureFlags:[]string(nil), csv:struct { files []string; ignorePatterns []string }{files:[]string(nil), ignorePatterns:[]string(nil)}, nvmllib:nvml.Interface(nil)}
        	            	actual  : generate.options{output:"", format:"yaml", deviceNameStrategies:[]string(nil), driverRoot:"/build/nvidia-container-toolkit/src/nvidia-container-toolkit-1.18.0/testdata/lookup/rootfs-1", devRoot:"", nvidiaCDIHookPath:"/build/nvidia-container-toolkit/src/nvidia-container-toolkit-1.18.0/build/nvidia-cdi-hook", ldconfigPath:"", mode:"nvml", vendor:"example.com", class:"device", configSearchPaths:[]string(nil), librarySearchPaths:[]string(nil), disabledHooks:[]string(nil), enabledHooks:[]string(nil), featureFlags:[]string(nil), csv:struct { files []string; ignorePatterns []string }{files:[]string(nil), ignorePatterns:[]string(nil)}, nvmllib:nvml.Interface(nil)}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -6,3 +6,3 @@
        	            	  devRoot: (string) "",
        	            	- nvidiaCDIHookPath: (string) (len=24) "/usr/bin/nvidia-cdi-hook",
        	            	+ nvidiaCDIHookPath: (string) (len=89) "/build/nvidia-container-toolkit/src/nvidia-container-toolkit-1.18.0/build/nvidia-cdi-hook",
        	            	  ldconfigPath: (string) "",
        	Test:       	TestGenerateSpec/default
=== RUN   TestGenerateSpec/disableHooks1
    generate_test.go:460:
        	Error Trace:	/build/nvidia-container-toolkit/src/nvidia-container-toolkit-1.18.0/cmd/nvidia-ctk/cdi/generate/generate_test.go:460
        	Error:      	Not equal:
        	            	expected: generate.options{output:"", format:"yaml", deviceNameStrategies:[]string(nil), driverRoot:"/build/nvidia-container-toolkit/src/nvidia-container-toolkit-1.18.0/testdata/lookup/rootfs-1", devRoot:"", nvidiaCDIHookPath:"/usr/bin/nvidia-cdi-hook", ldconfigPath:"", mode:"nvml", vendor:"example.com", class:"device", configSearchPaths:[]string(nil), librarySearchPaths:[]string(nil), disabledHooks:[]string{"enable-cuda-compat"}, enabledHooks:[]string(nil), featureFlags:[]string(nil), csv:struct { files []string; ignorePatterns []string }{files:[]string(nil), ignorePatterns:[]string(nil)}, nvmllib:nvml.Interface(nil)}
        	            	actual  : generate.options{output:"", format:"yaml", deviceNameStrategies:[]string(nil), driverRoot:"/build/nvidia-container-toolkit/src/nvidia-container-toolkit-1.18.0/testdata/lookup/rootfs-1", devRoot:"", nvidiaCDIHookPath:"/build/nvidia-container-toolkit/src/nvidia-container-toolkit-1.18.0/build/nvidia-cdi-hook", ldconfigPath:"", mode:"nvml", vendor:"example.com", class:"device", configSearchPaths:[]string(nil), librarySearchPaths:[]string(nil), disabledHooks:[]string{"enable-cuda-compat"}, enabledHooks:[]string(nil), featureFlags:[]string(nil), csv:struct { files []string; ignorePatterns []string }{files:[]string(nil), ignorePatterns:[]string(nil)}, nvmllib:nvml.Interface(nil)}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -6,3 +6,3 @@
        	            	  devRoot: (string) "",
        	            	- nvidiaCDIHookPath: (string) (len=24) "/usr/bin/nvidia-cdi-hook",
        	            	+ nvidiaCDIHookPath: (string) (len=89) "/build/nvidia-container-toolkit/src/nvidia-container-toolkit-1.18.0/build/nvidia-cdi-hook",
        	            	  ldconfigPath: (string) "",
        	Test:       	TestGenerateSpec/disableHooks1
=== RUN   TestGenerateSpec/disableHooks2
    generate_test.go:460:
        	Error Trace:	/build/nvidia-container-toolkit/src/nvidia-container-toolkit-1.18.0/cmd/nvidia-ctk/cdi/generate/generate_test.go:460
        	Error:      	Not equal:
        	            	expected: generate.options{output:"", format:"yaml", deviceNameStrategies:[]string(nil), driverRoot:"/build/nvidia-container-toolkit/src/nvidia-container-toolkit-1.18.0/testdata/lookup/rootfs-1", devRoot:"", nvidiaCDIHookPath:"/usr/bin/nvidia-cdi-hook", ldconfigPath:"", mode:"nvml", vendor:"example.com", class:"device", configSearchPaths:[]string(nil), librarySearchPaths:[]string(nil), disabledHooks:[]string{"enable-cuda-compat", "update-ldcache"}, enabledHooks:[]string(nil), featureFlags:[]string(nil), csv:struct { files []string; ignorePatterns []string }{files:[]string(nil), ignorePatterns:[]string(nil)}, nvmllib:nvml.Interface(nil)}
        	            	actual  : generate.options{output:"", format:"yaml", deviceNameStrategies:[]string(nil), driverRoot:"/build/nvidia-container-toolkit/src/nvidia-container-toolkit-1.18.0/testdata/lookup/rootfs-1", devRoot:"", nvidiaCDIHookPath:"/build/nvidia-container-toolkit/src/nvidia-container-toolkit-1.18.0/build/nvidia-cdi-hook", ldconfigPath:"", mode:"nvml", vendor:"example.com", class:"device", configSearchPaths:[]string(nil), librarySearchPaths:[]string(nil), disabledHooks:[]string{"enable-cuda-compat", "update-ldcache"}, enabledHooks:[]string(nil), featureFlags:[]string(nil), csv:struct { files []string; ignorePatterns []string }{files:[]string(nil), ignorePatterns:[]string(nil)}, nvmllib:nvml.Interface(nil)}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -6,3 +6,3 @@
        	            	  devRoot: (string) "",
        	            	- nvidiaCDIHookPath: (string) (len=24) "/usr/bin/nvidia-cdi-hook",
        	            	+ nvidiaCDIHookPath: (string) (len=89) "/build/nvidia-container-toolkit/src/nvidia-container-toolkit-1.18.0/build/nvidia-cdi-hook",
        	            	  ldconfigPath: (string) "",
        	Test:       	TestGenerateSpec/disableHooks2
=== RUN   TestGenerateSpec/disableHooksAll
    generate_test.go:460:
        	Error Trace:	/build/nvidia-container-toolkit/src/nvidia-container-toolkit-1.18.0/cmd/nvidia-ctk/cdi/generate/generate_test.go:460
        	Error:      	Not equal:
        	            	expected: generate.options{output:"", format:"yaml", deviceNameStrategies:[]string(nil), driverRoot:"/build/nvidia-container-toolkit/src/nvidia-container-toolkit-1.18.0/testdata/lookup/rootfs-1", devRoot:"", nvidiaCDIHookPath:"/usr/bin/nvidia-cdi-hook", ldconfigPath:"", mode:"nvml", vendor:"example.com", class:"device", configSearchPaths:[]string(nil), librarySearchPaths:[]string(nil), disabledHooks:[]string{"all"}, enabledHooks:[]string(nil), featureFlags:[]string(nil), csv:struct { files []string; ignorePatterns []string }{files:[]string(nil), ignorePatterns:[]string(nil)}, nvmllib:nvml.Interface(nil)}
        	            	actual  : generate.options{output:"", format:"yaml", deviceNameStrategies:[]string(nil), driverRoot:"/build/nvidia-container-toolkit/src/nvidia-container-toolkit-1.18.0/testdata/lookup/rootfs-1", devRoot:"", nvidiaCDIHookPath:"/build/nvidia-container-toolkit/src/nvidia-container-toolkit-1.18.0/build/nvidia-cdi-hook", ldconfigPath:"", mode:"nvml", vendor:"example.com", class:"device", configSearchPaths:[]string(nil), librarySearchPaths:[]string(nil), disabledHooks:[]string{"all"}, enabledHooks:[]string(nil), featureFlags:[]string(nil), csv:struct { files []string; ignorePatterns []string }{files:[]string(nil), ignorePatterns:[]string(nil)}, nvmllib:nvml.Interface(nil)}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -6,3 +6,3 @@
        	            	  devRoot: (string) "",
        	            	- nvidiaCDIHookPath: (string) (len=24) "/usr/bin/nvidia-cdi-hook",
        	            	+ nvidiaCDIHookPath: (string) (len=89) "/build/nvidia-container-toolkit/src/nvidia-container-toolkit-1.18.0/build/nvidia-cdi-hook",
        	            	  ldconfigPath: (string) "",
        	Test:       	TestGenerateSpec/disableHooksAll
=== RUN   TestGenerateSpec/enableChmodHook
    generate_test.go:460:
        	Error Trace:	/build/nvidia-container-toolkit/src/nvidia-container-toolkit-1.18.0/cmd/nvidia-ctk/cdi/generate/generate_test.go:460
        	Error:      	Not equal:
        	            	expected: generate.options{output:"", format:"yaml", deviceNameStrategies:[]string(nil), driverRoot:"/build/nvidia-container-toolkit/src/nvidia-container-toolkit-1.18.0/testdata/lookup/rootfs-1", devRoot:"", nvidiaCDIHookPath:"/usr/bin/nvidia-cdi-hook", ldconfigPath:"", mode:"management", vendor:"example.com", class:"device", configSearchPaths:[]string(nil), librarySearchPaths:[]string(nil), disabledHooks:[]string{"enable-cuda-compat", "update-ldcache", "disable-device-node-modification"}, enabledHooks:[]string{"chmod"}, featureFlags:[]string(nil), csv:struct { files []string; ignorePatterns []string }{files:[]string(nil), ignorePatterns:[]string(nil)}, nvmllib:nvml.Interface(nil)}
        	            	actual  : generate.options{output:"", format:"yaml", deviceNameStrategies:[]string(nil), driverRoot:"/build/nvidia-container-toolkit/src/nvidia-container-toolkit-1.18.0/testdata/lookup/rootfs-1", devRoot:"", nvidiaCDIHookPath:"/build/nvidia-container-toolkit/src/nvidia-container-toolkit-1.18.0/build/nvidia-cdi-hook", ldconfigPath:"", mode:"management", vendor:"example.com", class:"device", configSearchPaths:[]string(nil), librarySearchPaths:[]string(nil), disabledHooks:[]string{"enable-cuda-compat", "update-ldcache", "disable-device-node-modification"}, enabledHooks:[]string{"chmod"}, featureFlags:[]string(nil), csv:struct { files []string; ignorePatterns []string }{files:[]string(nil), ignorePatterns:[]string(nil)}, nvmllib:nvml.Interface(nil)}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -6,3 +6,3 @@
        	            	  devRoot: (string) "",
        	            	- nvidiaCDIHookPath: (string) (len=24) "/usr/bin/nvidia-cdi-hook",
        	            	+ nvidiaCDIHookPath: (string) (len=89) "/build/nvidia-container-toolkit/src/nvidia-container-toolkit-1.18.0/build/nvidia-cdi-hook",
        	            	  ldconfigPath: (string) "",
        	Test:       	TestGenerateSpec/enableChmodHook
--- FAIL: TestGenerateSpec (0.00s)
    --- FAIL: TestGenerateSpec/default (0.00s)
    --- FAIL: TestGenerateSpec/disableHooks1 (0.00s)
    --- FAIL: TestGenerateSpec/disableHooks2 (0.00s)
    --- FAIL: TestGenerateSpec/disableHooksAll (0.00s)
    --- FAIL: TestGenerateSpec/enableChmodHook (0.00s)
=== RUN   TestSplitOnAnnotation
=== RUN   TestSplitOnAnnotation/non-matching_annotation
=== RUN   TestSplitOnAnnotation/single_annotation_present
=== RUN   TestSplitOnAnnotation/non-matching_annotations_are_not_removed
=== RUN   TestSplitOnAnnotation/duplicated_annotations_different_names
=== RUN   TestSplitOnAnnotation/annotation_with_different_values
--- PASS: TestSplitOnAnnotation (0.00s)
    --- PASS: TestSplitOnAnnotation/non-matching_annotation (0.00s)
    --- PASS: TestSplitOnAnnotation/single_annotation_present (0.00s)
    --- PASS: TestSplitOnAnnotation/non-matching_annotations_are_not_removed (0.00s)
    --- PASS: TestSplitOnAnnotation/duplicated_annotations_different_names (0.00s)
    --- PASS: TestSplitOnAnnotation/annotation_with_different_values (0.00s)
FAIL
FAIL	github.com/NVIDIA/nvidia-container-toolkit/cmd/nvidia-ctk/cdi/generate	0.004s
?   	github.com/NVIDIA/nvidia-container-toolkit/cmd/nvidia-ctk/cdi/list	[no test files]
?   	github.com/NVIDIA/nvidia-container-toolkit/cmd/nvidia-ctk/cdi/transform	[no test files]
?   	github.com/NVIDIA/nvidia-container-toolkit/cmd/nvidia-ctk/cdi/transform/root	[no test files]
```

--- a/cmd/nvidia-ctk/cdi/generate/generate_test.go
+++ b/cmd/nvidia-ctk/cdi/generate/generate_test.go
@@ -450,6 +450,9 @@ containerEdits:
 	}
 
 	for _, tc := range testCases {
+		// Apply overrides for all test cases:
+		tc.options.nvidiaCDIHookPath = "/usr/bin/nvidia-cdi-hook"
+
 		t.Run(tc.description, func(t *testing.T) {
 			c := command{
 				logger: logger,

